docs/content/guides/dagster-pipes.mdx

```diff
@@ -0,0 +1,76 @@
+---
+title: Dagster Pipes | Dagster Docs
+description: "Dagster Pipes provides a protocol between the orchestration environment (Dagster) and external execution (ex: Databricks) and a toolkit for building implementations of that protocol."
+---
+
+# Dagster Pipes
+
+Dagster Pipes is a toolkit for building integrations between Dagster and external execution environments. It standardizes the process of passing parameters, injecting context information, ingesting logs, and collecting metadata all while remaining agnostic to how remote computations are launched in those environments. This enables the separation of orchestration and business logic in the Dagster ecosystem.
+
+It also smooths the process of incorporating pre-existing code, business logic, and execution environments into Dagster. Using Dagster Pipes, you add just a few lines to your code and it becomes seamlessly invocable by the orchestrator. In turn, you can stream logs and metadata back to Dagster so you can leverage its observability, lineage, cataloging, and debugging capabilities.
+
+---
+
+## Benefits
+
+With Dagster Pipes, you can:
+
+- Incorporate existing code into Dagster without huge refactors
+- Onboard stakeholder teams onto Dagster incrementally
+- Run code in external environments, and:
+  - Easily pass parameters to the code
+  - Stream unstructured logs and structured metadata back to Dagster
+- Separate orchestration and business logic environments
+- Use languages other than Python with Dagster
+
+---
+
+## Limitations
+
+While Dagster Pipes is lightweight and flexible, there are a few limitations to be aware of:
+
+- **Step launchers are currently incompatible.**
```

Comment:
> **Step launchers are currently incompatible.**

I don't think is an accurate framing. Pipes are a lightweight _alternative_ to step launchers.

Reply:
That's fair. Would you agree that it's accurate to portray this as an either-or situation - that is, you can use step launchers, and you could use pipes, but not necessarily together?

Reply:
Definitely either-or.